### ISSUE TITLE
When navigating AmazonECSClient.java example source file, some definitions fail to load.

### ISSUE THREAD

**Issue by shurcooL** [sourcegraph/sourcegraph.com#388]
For example, if you go to https://sourcegraph.com/github.com/aws/aws-sdk-java@dbfa6d6f2285ca088e60cf5c01ba8d93a26d1c2c/.tree/aws-java-sdk-ecs/src/main/java/com/amazonaws/services/ecs/AmazonECSClient.java#startline=271&endline=271 and click on `ExecutionContext` type or `createExecutionContext` function, an error occurs and the definition fails to load:

![image](https://cloud.githubusercontent.com/assets/1924134/7400389/75490b2c-ee71-11e4-80e1-4cf517f700d5.png)

HT @neelance.

**Comment by neelance** [comment]
In general, wouldn't it be better to show "No definition available" than this scary error box? ;-) Then the user would think "Okay, for some reason it does not work in this case" instead of "Oh, seems like their system is broken". Just to keep appearances, of course the problem needs to be logged and resolved eventually.

**Comment by shurcooL** [comment]
You're right, I agree. It only affects some identifiers which seem to be missing a definition, many others work well.

It's already handled well by the tooltip - it simply doesn't show up when the definition is missing.

I think the best way to "resolve" the issue would be to avoid linking/highlighting defs that are missing a definition, since clicking them (or hovering over them) would not succeed anyway. But they are highlighted and linked, which is the problem that we need to resolve.
